kayenta-web/src/main/java/com/netflix/kayenta/controllers/CanaryConfigController.java

```diff
@@ -111,14 +111,26 @@ public CanaryConfigUpdateResponse storeCanaryConfig(
     canaryConfig.setUpdatedTimestampIso(
         Instant.ofEpochMilli(canaryConfig.getUpdatedTimestamp()).toString());
 
-    String canaryConfigId = UUID.randomUUID() + "";
+    String canaryConfigId;
+
+    if (StringUtils.isEmpty(canaryConfig.getId())) {
+      canaryConfigId = UUID.randomUUID() + "";
+
+      // Ensure that the canary config id is stored within the canary config itself.
+      if (StringUtils.isEmpty(canaryConfig.getId())) {
+        canaryConfig = canaryConfig.toBuilder().id(canaryConfigId).build();
```

Comment:
Don't you have an `@Setter`?

Reply:
Shouldn't, will remove it. Trying to stick to not modifying the instance once it's constructed.